### PR TITLE
Finish implementation of /formatshelp

### PR DESCRIPTION
**Pull request by Slayer95**
Finish implementation of /formatshelp
- Now mostly merged with /tiers and /othermetas.
- Format descriptions are arrays of HTML strings, which are joined to be displayed.

@TheImmortal @DanUgore

### CODE REVIEW

**Comment by Slayer95** [comment]
Screenies: http://imgur.com/6vdaxR2,CEDxI7l

File: chat-plugins/info.js [hunk 1]
```diff
@@ -1833,145 +1833,84 @@ var commands = exports.commands = {
...
-			buf.push({
-				name: format.name,
-				gameType: format.gameType || 'singles',
-				mod: format.mod,
-				searchShow: format.searchShow,
-				desc: format.desc || 'No description.'
-			});
+		for (var sectionId in sections) {
+			buf.push("<h3>" + Tools.escapeHTML(sections[sectionId].name) + "</h3>");
+			buf.push("<table border=\"2\" cellspacing =\"0\" cellpadding=\"5\"><thead><th>Name</th><th>Mode</th><th>Description</th></thead><tbody>");
```
Comment (Zarel) [inline chat-plugins/info.js]: > `border="2"`

Are you _trying_ to make it look ugly? :P

Comment (Slayer95) [inline chat-plugins/info.js]: I use that border in all my tables :[

What do you suggest?

Comment (Rhythms88) [inline chat-plugins/info.js]: I don't think the space after "cellspacing" is needed

Correct me if I'm wrong :P

**Comment by TheImmortal** [comment]
Hmm, it was discussed before (I think with @xfix) that `/formats` should be linking to the format page on the Smogon Dex. Like `/format ou` and `/format 1v1`, for example, so it works for both standard tiers and OMs.

How does it look with the OM descriptions? I doubt they look good in a table. Also, I don't even think the table looks nice.

Anyway, personally, I like `tiers` and `om` the way they are right now. The only change that needs to happen is `/formatshelp` and `/formats` should be linking to the formats page on the Smogon Dex.

File: chat-plugins/info.js (hunk 1 above)
Comment (Zarel) [inline chat-plugins/info.js]: 1px gray border is a good start, that'll make it look like it's from 10 years ago instead of 20 years ago.

**Comment by Slayer95** [comment]
http://i.imgur.com/Kvk1OBb.png

I suppose it's not an artwork, but I find it useful. And if the Smogdex links were added it would be even more useful.
As for a command restricted to retrieving the Smogon formats page, that's ``/smogdex``

**Comment by TheImmortal** [comment]
Also, is there still a table for displaying on format? Show me `!om 1v1` and `!tier ou`. If that looks fine then I guess there's no harm in the rest of the changes because that's all that really matters to me.

**Comment by Slayer95** [comment]
http://i.imgur.com/II7qGaR.png
OU yields currently an interesting result, but I can probably change it to display the exact match.
Do you want this table gone for unique-matches. @TheImmortal?

Also, updated style, @Zarel, though we should probably have a class client-side for this.

**Comment by TheImmortal** [comment]
Yeah remove it for single matches please...

**Comment by Slayer95** [comment]
Here: http://i.imgur.com/R3FR8re.png

File: chat-plugins/info.js [hunk 2]
```diff
@@ -1833,145 +1833,95 @@ var commands = exports.commands = {
...
+			var format = Tools.getFormat(Object.values(sections)[0].formats[0]);
+			return this.sendReplyBox(format.desc.join("<br />"));
+		}
+
+		// Build tables
+		var buf = [];
+		for (var sectionId in sections) {
+			if (exactMatch && sectionId !== exactMatch) continue;
+			buf.push("<h3>" + Tools.escapeHTML(sections[sectionId].name) + "</h3>");
+			buf.push("<table style=\"border:1px solid gray\" cellspacing =\"0\" cellpadding=\"5\"><thead><th style=\"border:1px solid gray\" >Name</th><th style=\"border:1px solid gray\" >Mode</th><th style=\"border:1px solid gray\" >Description</th></thead><tbody>");
```
Comment (Zarel) [inline chat-plugins/info.js]: Blah, you're going to need to set some sort of border-collapse option if you want this to work correctly...

**Comment by Slayer95** [comment]
Looks sober with ``border-collapse:collapse``. Now this looks good imo (http://imgur.com/ynA6OoK)

**Pull request merged**